x/derivatives/keeper/perpetual_futures_test.go

```diff
@@ -16,6 +16,7 @@ func (suite *KeeperTestSuite) TestOpenPerpetualFuturesPosition() {
 		QuoteDenom: "uusdc",
 	}
 
+	// TODO: add failure case due to the lack of the available asset in the pool
```

Comment:
Can you implement all TODOs on this PR?

Reply:
I asked @Senna46 to implement in the other branch.
So, please skip now.